File: mobile/library/swift/src/filters/ResponseFilterCallbacksImpl.swift
@_implementationOnly import EnvoyEngine
import Foundation

/// Envoy implementation of ResponseFilterCallbacks
final class ResponseFilterCallbacksImpl: NSObject {
  private let callbacks: EnvoyHTTPFilterCallbacks

  init(callbacks: EnvoyHTTPFilterCallbacks) {
    self.callbacks = callbacks
    super.init()
  }
}

extension ResponseFilterCallbacksImpl: ResponseFilterCallbacks {
  func resumeResponse() {
    self.callbacks.resumeIteration()
  }
}

File: mobile/library/kotlin/io/envoyproxy/envoymobile/filters/ResponseFilterCallbacksImpl.kt
package io.envoyproxy.envoymobile

import io.envoyproxy.envoymobile.engine.types.EnvoyHTTPFilterCallbacks

/**
 * Envoy implementation of `ResponseFilterCallbacks`.
 */
internal class ResponseFilterCallbacksImpl constructor(
  internal val callbacks: EnvoyHTTPFilterCallbacks
) : ResponseFilterCallbacks {

  override fun resumeResponse() {
    callbacks.resumeIteration()
  }

  override fun resetIdleTimer() {
    callbacks.resetIdleTimer()
  }
}
